### ISSUE TITLE
Command Option Syntax Is Confusing

### ISSUE THREAD

**Issue by CraigMiloRogers** [usc-isi-i2/kgtk#74]
Users are confused about where to put positional arguments, typically file names.  If they place them after an option that takes a varying number of arguments, such as a column list, the file name is absorbed into the list.

**Comment by CraigMiloRogers** [comment]
Proposal:  Use -i (--input-file) and -o (--output-file) instead of positional arguments.

**Comment by dgarijo** [comment]
I have no preference, as long as it is consistent across commands.

**Comment by szeke** [comment]
I agree with the suggestion to use options to specify input files so it doesn't matter where it occurs.

I suppose that commands that take different  types of files as input (eg join), will use the standard input as the `-I` file. If there are multiple input files for `-I`, do we repeat the option? `-i file -i file`. If in addition a file is given in stdin, is there a way to specify where it goes in the options? `-i file -i - -i file`?

**Comment by dgarijo** [comment]
I would have two different commands:

`-i` (--input) for a simple file

`-is` (--inputs) for multiple inputs.

That way you could have `command -i input1.txt`
or
`command -is input1 input2 input3`

**Comment by szeke** [comment]
I think the convention is that single dash options have to be one character. I think the decision is whether multiple values to options are a list or we list the option multiple times.

**Comment by CraigMiloRogers** [comment]
If the input files serve different purposes, such as left/right or input/filter, I prefer to use special options.
If we are inputting multiple files to be processed as a series, listing the option once with multiple files after it appears to be the only choice with Python argparse.

**Comment by CraigMiloRogers** [comment]
Nevermind, I found the example in the documentation.  The key to allowing multiple "-i" options is action="append".

**Comment by CraigMiloRogers** [comment]
We've updated the commands and can migrate to prohibit positional arguments.

**Comment by CraigMiloRogers** [comment]
This issue is waiting for our decision to prohibit positional arguments.
